src/test/java/picard/sam/markduplicates/MarkDuplicatesSetSizeHistogramTester.java

```diff
@@ -0,0 +1,128 @@
+
+                    System.err.println("Mismatching read:");
+                    System.err.println(record.getSAMString());
+                }
+                Assert.assertEquals(record.getDuplicateReadFlag(), value);
+            }
+        }
+
+        // Ensure the program output the same number of records as were read in
+        Assert.assertEquals(outputRecords, this.getNumberOfRecords(), ("saw " + outputRecords + " output records, vs. " + this.getNumberOfRecords() + " input records"));
+
+        // Check the values written to metrics.txt against our input expectations
+        final MetricsFile<DuplicationMetrics, Double> metricsOutput = new MetricsFile<DuplicationMetrics, Double>();
+        try {
+            metricsOutput.read(new FileReader(metricsFile));
+        } catch (final FileNotFoundException ex) {
+            System.err.println("Metrics file not found: " + ex);
+        }
+        Assert.assertEquals(metricsOutput.getMetrics().size(), 1);
+        final DuplicationMetrics observedMetrics = metricsOutput.getMetrics().get(0);
+        Assert.assertEquals(observedMetrics.UNPAIRED_READS_EXAMINED, expectedMetrics.UNPAIRED_READS_EXAMINED, "UNPAIRED_READS_EXAMINED does not match expected");
+        Assert.assertEquals(observedMetrics.READ_PAIRS_EXAMINED, expectedMetrics.READ_PAIRS_EXAMINED, "READ_PAIRS_EXAMINED does not match expected");
+        Assert.assertEquals(observedMetrics.UNMAPPED_READS, expectedMetrics.UNMAPPED_READS, "UNMAPPED_READS does not match expected");
+        Assert.assertEquals(observedMetrics.UNPAIRED_READ_DUPLICATES, expectedMetrics.UNPAIRED_READ_DUPLICATES, "UNPAIRED_READ_DUPLICATES does not match expected");
+        Assert.assertEquals(observedMetrics.READ_PAIR_DUPLICATES, expectedMetrics.READ_PAIR_DUPLICATES, "READ_PAIR_DUPLICATES does not match expected");
+        Assert.assertEquals(observedMetrics.READ_PAIR_OPTICAL_DUPLICATES, expectedMetrics.READ_PAIR_OPTICAL_DUPLICATES, "READ_PAIR_OPTICAL_DUPLICATES does not match expected");
+        Assert.assertEquals(observedMetrics.PERCENT_DUPLICATION, expectedMetrics.PERCENT_DUPLICATION, "PERCENT_DUPLICATION does not match expected");
+        Assert.assertEquals(observedMetrics.ESTIMATED_LIBRARY_SIZE, expectedMetrics.ESTIMATED_LIBRARY_SIZE, "ESTIMATED_LIBRARY_SIZE does not match expected");
+        Assert.assertEquals(observedMetrics.SECONDARY_OR_SUPPLEMENTARY_RDS, expectedMetrics.SECONDARY_OR_SUPPLEMENTARY_RDS, "SECONDARY_OR_SUPPLEMENTARY_RDS does not match expected");
+
+
+        // Check contents of set size bin against expected values //
+        for (final Histogram<Double> histo : metricsOutput.getAllHistograms()) {
+            final String label = histo.getValueLabel();
+            for (final Double bin : histo.keySet()) {
+                final String binStr = String.valueOf(bin);
+                final List<String> labelBinStr = Arrays.asList(label, binStr);
+                if (expectedSetSizeMap.containsKey(labelBinStr)) {
```

Comment:
This will pass even if this `containsKey()` never returns true in this loop. Is that correct? If not, the test should check to see that at least one (?) label matches.